pkg/catalog/manager/manager.go

```diff
@@ -53,3 +59,40 @@ func (m *Manager) deleteChart(toDelete string, namespace string) error {
 	}
 	return nil
 }
+
+func (m *Manager) DeleteOldTemplateContent() {
+	templateContents, err := m.templateContentClient.List(metav1.ListOptions{})
+	if err != nil {
+		logrus.Error(err)
```

Comment:
What happens if there is an error? will the code never be executed until next restart? Should we retry this if error happens?

Reply:
The code won't be called again until next restart, but afaik it shouldn't fail unless the k8s api server is unavailable. My bigger concern is for the memory & time consumption of this method.

Reply:
But we should still retry this method until it succeeds. There are plenty of reasons k8s api call could fail. Otherwise the cache will never be cleaned up.